.circleci/config-2_1.yml.high_res.patch

```diff
@@ -1,34 +1,79 @@
-22,23c22,23
-<     #exec_resource_class: xlarge
-<   parallelism: 4
----
->     exec_resource_class: xlarge
->   parallelism: 100
-28,29c28,29
-<     #exec_resource_class: xlarge
-<   parallelism: 1
----
->     exec_resource_class: xlarge
->   parallelism: 5
-34,35c34,35
-<     #exec_resource_class: xlarge
-<   parallelism: 1
----
->     exec_resource_class: xlarge
->   parallelism: 2
-40c40
-<     #exec_resource_class: xlarge
----
->     exec_resource_class: xlarge
-46,47c46,47
-<     #exec_resource_class: xlarge
-<   parallelism: 4
----
->     exec_resource_class: xlarge
->   parallelism: 100
-52,53c52,53
-<     #exec_resource_class: xlarge
-<   parallelism: 1
----
->     exec_resource_class: xlarge
->   parallelism: 2
+@@ -19,39 +19,51 @@ default_env_vars: &default_env_vars
+ j8_par_executor: &j8_par_executor
+   executor:
+     name: java8-executor
+-    #exec_resource_class: xlarge
+-  parallelism: 4
++    exec_resource_class: xlarge
++  parallelism: 100
+ 
+ j8_small_par_executor: &j8_small_par_executor
+   executor:
+     name: java8-executor
+-    #exec_resource_class: xlarge
++    exec_resource_class: xlarge
++  parallelism: 5
++
++j8_small_executor: &j8_small_executor
++  executor:
++    name: java8-executor
++    exec_resource_class: medium
+   parallelism: 1
+ 
+ j8_medium_par_executor: &j8_medium_par_executor
+   executor:
+     name: java8-executor
+-    #exec_resource_class: xlarge
+-  parallelism: 1
++    exec_resource_class: xlarge
++  parallelism: 2
+ 
+ j8_seq_executor: &j8_seq_executor
+   executor:
+     name: java8-executor
+-    #exec_resource_class: xlarge
++    exec_resource_class: xlarge
+   parallelism: 1 # sequential, single container tests: no parallelism benefits
+ 
+ j11_par_executor: &j11_par_executor
+   executor:
+     name: java11-executor
+-    #exec_resource_class: xlarge
+-  parallelism: 4
++    exec_resource_class: xlarge
++  parallelism: 100
+ 
+-j11_small_par_executor: &j11_small_par_executor
++j11_small_executor: &j11_small_executor
+   executor:
+     name: java11-executor
+-    #exec_resource_class: xlarge
++    exec_resource_class: medium
+   parallelism: 1
+ 
++j11_small_par_executor: &j11_small_par_executor
++  executor:
++    name: java11-executor
++    exec_resource_class: xlarge
++  parallelism: 2
++
```

Comment:
Below stuff shouldn't appear here. 
+j11_small_par_executor: &j11_small_par_executor
+  executor:
+    name: java11-executor
+    exec_resource_class: xlarge
+  parallelism: 2

The flow is as follows:
1) Because we need to add new jobs, we do that in both config.yml and config-2_1.yml
2) The jobs are added to config.yml and config-2_1.yml with their low res and commit those two files the way the lowres is supposed to look like
3) Then the patch creation comes:
 - update the config-2_1.yml the way you want it for midres and create a patch with name config-2_1.yml.mid_res.patch. When you look at the patch, only resources will be changed - from those used for lowres to those used for highres. Save the patch and revert the changes in config-2_1.yml so it stays in the lowres
- repeat the procedure for highres patch
4) Now when you have your patch files, you don't have to do anything else then run generate.sh which will update config.yml.LOWRES, config.yml.MIDRES, config.yml.HIGHRES

Reply:
I am not following you here :-) on point 3.1 you are dealing with midres but you mention highres...?

I _think_ your concern is with 'adding' `j11_small_par_executor`? If you look a bit higher in the patch you will see this is `diff` noise. Diff gets confused between `j11_small_executor` (the new one) and `j11_small_par_executor`. In fact I am only adding `j11_small_executor` but the diff makes a mess of that.

I am following what I see on the original midres patch i.e. which is where new executors are [added](https://github.com/apache/cassandra/blob/be97d67c0d3bad9e35eb4a5b4476a49e1ef8c935/.circleci/config-2_1.yml.mid_res.patch#L37)

Also if you run `generate.sh` you'll see it runs and files don't change. So everything is in sync imo.

Am I barking up the wrong tree?

Reply:
It was easier for me to make a clone in my repo. 
So looking here, based on your branch I played only with the order of executors, etc
https://github.com/apache/cassandra/compare/trunk...ekaterinadimitrova2:CASSANDRA-16121
I created only the LOWRES and MIDRES, j8_small_executor is better to be introduced first in lowres as it is lowres that gets propagated to all three config options. I know you will ask me then why in the previous update I introduced the large and very large executors in midres - the response is because they are used only there for our needs.
Hope now it is more clear. Feel free to cherry pick from my branch if you agree with me.

Reply:
@ekaterinadimitrova2 thx for looking into this but your patch doesn't apply unfortunatelly :-( I think there are 2 patches in [one](https://github.com/apache/cassandra/compare/trunk...ekaterinadimitrova2:CASSANDRA-16121#diff-ba268153edd6228316bb0f7b9e21e600R236) file?

I take it you'd rather have the small executors defined in the original config21.yml and let that propagate? I can do that but the current SHA has been tested low, mid and high. That change would mean retesting that again. Do you feel this is a cosmetic nit or do you feel stronly about it?

Reply:
The midres created was able to apply for me... Not sure what is the issue on your end. Maybe you tried the generate.sh script which looks also at the highres and I made a POV. only for midres.
Yes, the small executors should be there, that's what I meant. 
Unfortunately, I think it should be moved. Otherwise we are not consistent and it becomes more confusing in the future for updates. All executors which are propagated in the three configurations and have lowres version are defined in config-2_1.yml

Reply:
weird. I get `patch -o config-2_1.yml.MIDRES config-2_1.yml config-2_1.yml.mid_res.patch
patching file config-2_1.yml.MIDRES (read from config-2_1.yml)
patching file config-2_1.yml.MIDRES (read from config-2_1.yml)
Reversed (or previously applied) patch detected!  Assume -R? [n]`

Anyway I can do that change tomorrow